### PR TITLE
- Remove the login landing page

### PR DESCRIPTION
**Pull request by jennifer-gan**
- [x] Remove the login landing page
- [x] Remove the back button as it is not needed anymore
- [x] Adjust layout due to the removal of back button and remove the propTypes 
requirement for onBack function

as seen below

[Screencast from 2024-12-10 11-19-04.webm](https://github.com/user-attachments/assets/6400ed74-f0c4-46ae-bb22-54424a5999e2)

### CODE REVIEW

File: Client/src/Pages/Auth/Login.jsx [hunk 1]
```diff
@@ -19,99 +19,6 @@ import { logger } from "../../Utils/Logger";
 import "./index.css";
```
Comment (ajhollid) [inline Client/src/Pages/Auth/Login.jsx:19]: `import Mail from "../../assets/icons/mail.svg?react";`

This icon import can be removed, it is no longer used

File: Client/src/Pages/Auth/Login.jsx [hunk 2]
```diff
@@ -168,26 +75,8 @@ const StepOne = ({ form, errors, onSubmit, onChange, onBack }) => {
 					/>
 					<Stack
 						direction="row"
-						justifyContent="space-between"
+						justifyContent="flex-end"
 					>
-						<Button
-							variant="outlined"
-							color="info"
-							onClick={onBack}
```
Comment (ajhollid) [inline Client/src/Pages/Auth/Login.jsx:176]: `onBack` is no longer used, it can be removed from `StepOne` props

**Review by ajhollid — CHANGES_REQUESTED** [review]
Looks good to me!  Some unused imports and props dangling around that can be removed and we're good to go!

File: Client/src/Pages/Auth/Login.jsx [hunk 3]
```diff
@@ -384,7 +272,7 @@ const Login = () => {
 		password: DEMO !== undefined ? "Demouser1!" : "",
 	});
 	const [errors, setErrors] = useState({});
-	const [step, setStep] = useState(0);
+	const [step, setStep] = useState(1);
```
Comment (ajhollid) [inline Client/src/Pages/Auth/Login.jsx:275]: Let's stick with 0 indexing as everything else is 0 indexed, we should be consistent.

File: Client/src/Pages/Auth/Login.jsx [hunk 4]
```diff
@@ -533,15 +421,12 @@ const Login = () => {
 					},
 				}}
 			>
-				{step === 0 ? (
-					<LandingPage onContinue={() => setStep(1)} />
-				) : step === 1 ? (
+				{step === 1 ? (
```
Comment (ajhollid) [inline Client/src/Pages/Auth/Login.jsx:424]: This can be changed to 0 to maintain 0 indexing

File: Client/src/Pages/Auth/Login.jsx [hunk 5]
```diff
@@ -533,15 +421,12 @@ const Login = () => {
...
 					<StepOne
 						form={form}
 						errors={errors}
 						onSubmit={handleSubmit}
 						onChange={handleChange}
-						onBack={() => setStep(0)}
 					/>
 				) : (
 					step === 2 && (
```
Comment (ajhollid) [inline Client/src/Pages/Auth/Login.jsx:432]: Similarly this can be changed to 1

**Review by ajhollid — CHANGES_REQUESTED** [review]
Just noticed the indexing for steps, lets change it back to 0 indexing as everything else in the app is 0 indexed.

File: Client/src/Pages/Auth/Login.jsx (hunk 3 above)
Comment (jennifer-gan) [reply Client/src/Pages/Auth/Login.jsx]: Fixed in: https://github.com/bluewave-labs/checkmate/commit/76bfb7717ac3eada7d94e11a4ce9291119b45081

File: Client/src/Pages/Auth/Login.jsx (hunk 5 above)
Comment (jennifer-gan) [reply Client/src/Pages/Auth/Login.jsx]: This one was removed

File: Client/src/Pages/Auth/Login.jsx (hunk 4 above)
Comment (jennifer-gan) [reply Client/src/Pages/Auth/Login.jsx]: Fixed in : https://github.com/bluewave-labs/checkmate/commit/76bfb7717ac3eada7d94e11a4ce9291119b45081

File: Client/src/Pages/Auth/Login.jsx (hunk 2 above)
Comment (jennifer-gan) [reply Client/src/Pages/Auth/Login.jsx:176]: Fixed in: https://github.com/bluewave-labs/checkmate/commit/ba8cc04f2d58cfc0320be926d4af5776a70f7b80

File: Client/src/Pages/Auth/Login.jsx [hunk 6]
```diff
@@ -12,106 +12,12 @@ import TextInput from "../../Components/Inputs/TextInput";
...
-						onClick={onContinue}
-						sx={{
-							width: "100%",
-							"& svg": {
-								mr: theme.spacing(4),
-								"& path": {
-									stroke: theme.palette.other.icon,
-								},
-							},
-							"&:focus-visible": {
```
Comment (marcelluscaio) [inline Client/src/Pages/Auth/Login.jsx:56]: Maybe we shouldn't ditch all of the initial greeting. We should check that with @gorkem-bwl . I'd keep
- We are pleased to see you again!
- By continuing, you agree to our Terms of Service and Privacy Policy.

File: Client/src/Pages/Auth/Login.jsx [hunk 7]
```diff
@@ -433,9 +320,9 @@ const Login = () => {
 				setErrors((prev) => ({ ...prev, email: error.details[0].message }));
 				createToast({ body: error.details[0].message });
 			} else {
-				setStep(2);
```
Comment (marcelluscaio) [inline Client/src/Pages/Auth/Login.jsx:436]: @ajhollid I know you requested this change, but this gets a bit confusing to me: we are on the component StepOne, but it is  the step 0. If we are changing to step 0 I would suggest changing the name of the component as well

**Review by marcelluscaio — CHANGES_REQUESTED** [review]
Great, @shemy-gan ! On this one it would be a good opportunity to refactor and apply our rule of not having more than one component per file. Step One and Two could be their own component in their own file. The folder would be Login/components/StepOne and .../StepTwo

File: Client/src/Pages/Auth/Login.jsx (hunk 7 above)
Comment (ajhollid) [reply Client/src/Pages/Auth/Login.jsx:436]: > @ajhollid I know you requested this change, but this gets a bit confusing to me: we are on the component StepOne, but it is the step 0. If we are changing to step 0 I would suggest changing the name of the component as well

Agreed, it is confusing.  I agree changing the name of the component is a good idea anyways, `StepOne` is really not descriptive as to what Step One actually is.

@jennifer-gan can you please change the names when refactoring out the components?  Thank you!

Comment (jennifer-gan) [reply Client/src/Pages/Auth/Login.jsx:436]: sure, fixed here: https://github.com/bluewave-labs/checkmate/commit/c72334253d9bc4db2cf4daf900c975b884984e9e

**Review by ajhollid — APPROVED** [review]
Looks good to me, thanks for refactoring those components

**Pull request merged**